build-tools/automation/yaml-templates/run-nunit-tests.yaml

```diff
@@ -14,7 +15,11 @@ steps:
     if ([Environment]::OSVersion.Platform -eq "Unix") {
         if ("${{ parameters.useDotNet }}" -eq $true) {
             Write-Host '##vso[task.setvariable variable=TestResultsFormat]VSTest'
-            & $env:HOME/Library/Android/dotnet/dotnet test ${{ parameters.testAssembly }} --results-directory . --logger "trx;LogFileName=${{ parameters.testResultsFile }}" ${{ parameters.dotNetTestExtraArgs }} -- NUnit.NumberOfTestWorkers=${{ parameters.workers }}
+            $dotnetPath = "$env:HOME/Library/Android/dotnet/dotnet"
+            if ("${{ parameters.useHostedDotNetPath }}" -ne $true) {
+                $dotnetPath = "$env:HOME/android-toolchain/dotnet/dotnet"
+            }
+            & $dotnetPath test ${{ parameters.testAssembly }} --results-directory . --logger "trx;LogFileName=${{ parameters.testResultsFile }}" ${{ parameters.dotNetTestExtraArgs }} -- NUnit.NumberOfTestWorkers=${{ parameters.workers }}
```

Comment:
Could we change this to use `Test-Path` and we wouldn't need this new `useHostedDotNetPath` parameter?

```powershell
if (-not (Test-Path $dotnetPath)) {
  # Change the path
}
```

On Linux, would this also work to just use `dotnet` instead of `"$env:HOME/android-toolchain/dotnet/dotnet"`?

Reply:
We need to use the sandbox `dotnet` location, and the `dotnet` in path will be from the 5.0.x install that happens earlier in the build. I like the Test-Path plan though.